ts/components/services/ServicesTab.tsx

```diff
@@ -0,0 +1,271 @@
+import { GlobalState } from "../../store/reducers/types";
+import {
+  UserMetadata,
+  userMetadataSelector
+} from "../../store/reducers/userMetadata";
+import { makeFontStyleObject } from "../../theme/fonts";
+import customVariables from "../../theme/variables";
+import { getLogoForOrganization } from "../../utils/organizations";
+import { isTextIncludedCaseInsensitive } from "../../utils/strings";
+import ChooserListContainer from "../ChooserListContainer";
+import { withLightModalContext } from "../helpers/withLightModalContext";
+import { LightModalContextInterface } from "../ui/LightModal";
+import OrganizationLogo from "./OrganizationLogo";
+import ServicesSectionsList from "./ServicesSectionsList";
+
+type OwnProps = Readonly<{
+  heading: string;
+  isLocal?: boolean;
+  updateToast?: () => void;
+  sections: ReadonlyArray<ServicesSectionState>;
+  isRefreshing: boolean;
+  onRefresh: (hideToast?: boolean) => void; // tslint:disable-line bool-param-default
+  onServiceSelect: (service: ServicePublic) => void;
+  handleOnLongPressItem: () => void;
+  isLongPressEnabled: boolean;
+  updateOrganizationsOfInterestMetadata?: (
+    selectedItemIds: Option<Set<string>>
+  ) => void;
+  onItemSwitchValueChanged: (service: ServicePublic, value: boolean) => void;
+  tabOffset: Animated.Value;
+}>;
+
+type Props = OwnProps &
+  ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps> &
+  LightModalContextInterface;
+
+const styles = StyleSheet.create({
+  activeTextStyle: {
+    ...makeFontStyleObject(Platform.select, "600"),
+    fontSize: Platform.OS === "android" ? 16 : undefined,
+    fontWeight: Platform.OS === "android" ? "normal" : "bold"
+  },
+  textStyle: {
+    fontSize: customVariables.fontSizeSmall
+  },
+  organizationLogo: {
+    marginBottom: 0
+  },
+  icon: {
+    paddingHorizontal: (24 - 17) / 2 // (io-right icon width) - (io-trash icon width)
+  }
+});
+
+function renderOrganizationLogo(organizationFiscalCode: string) {
+  return (
+    <OrganizationLogo
+      logoUri={getLogoForOrganization(organizationFiscalCode)}
+      imageStyle={styles.organizationLogo}
+    />
+  );
+}
+
+function organizationContainsText(item: Organization, searchText: string) {
+  return isTextIncludedCaseInsensitive(item.name, searchText);
+}
+
+class ServicesTab extends React.PureComponent<Props> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {};
+    this.onSaveAreasOfInterest = this.onSaveAreasOfInterest.bind(this);
+    this.showChooserAreasOfInterestModal = this.showChooserAreasOfInterestModal.bind(
+      this
+    );
+    this.renderLocalQuickSectionDeletion = this.renderLocalQuickSectionDeletion.bind(
+      this
+    );
+  }
+  /**
+   * For tab Locals
+   */
+  private showChooserAreasOfInterestModal() {
+    const {
+      selectableOrganizations,
+      hideModal,
+      selectedOrganizations
+    } = this.props;
+
+    const OrganizationsList = createFactory(
```

Comment:
This could be moved outside from the class.
In this position it will be called each time _showChooserAreasOfInterestModal_ is fired

Reply:
shifted within https://github.com/teamdigitale/io-app/pull/1409/commits/0d0048cf3f60a0289763196f8f0a087155a75d81